### PR TITLE
Remove CUSTOM_PROJECT_MANIPULATOR_PARAMETERS, endpoint returns JSON e…

### PR DESCRIPTION
**Pull request by dominikbrazdil**
…xcept of String

### Checklist:

* [x] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [x] Have you added unit tests for your change?

### CODE REVIEW

**Comment by jbartece** [comment]
@janinko @alexcreasy  What is your opinion on that?

File: dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+package org.jboss.pnc.dto.response;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+
+/**
+ * // Default parameters for build configuration for a specific build type
```
Comment (michalovjan) [inline dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java]: "//" could be deleted

File: dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * @author dbrazdil
+ * @since 4.0
```
Comment (michalovjan) [inline dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java]: From where did Intellij get this number?

**Comment by jbartece** [comment]
@michalovjan I don't think it makes nicer code - it just complicates it. Another thing is that more important is, if it makes sense from the API perspective. I don't think so.

File: dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java (hunk 2 above)
Comment (jbartece) [reply dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java]: I am curious about that as well :-D

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildConfigurationEndpointTest.java [hunk 3]
```diff
@@ -482,9 +483,9 @@ public void testGetBuildTypeDefaultAlignmentParameters() throws RemoteResourceEx
         BuildConfigurationClient client = new BuildConfigurationClient(RestClientConfiguration.asAnonymous());
 
         for (BuildType buildType : BuildType.values()) {
-            Optional<String> params = client.getBuildTypeDefaultAlignmentParameters(buildType.name());
-            assertThat(params).isPresent();
-            assertThat(!params.get().isEmpty());
+            AlignmentParameters params = client.getBuildTypeDefaultAlignmentParameters(buildType.name());
+            assertThat(!params.getParameters().isEmpty());
+            assertThat(!params.getBuildType().isEmpty());
```
Comment (janinko) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildConfigurationEndpointTest.java]: I think it should be like this?
```suggestion
            assertThat(params.getParameters()).isNotEmpty();
            assertThat(params.getBuildType()).isNotEmpty();
```

**Comment by janinko** [comment]
@jbartece we should be returning JSON objects in responses.

Returning the `buildType` is what I advised Dominik as I thought it would make the returned object self-describing without the need to remember the request parameter. But I don't have strong opinion on including that parameter.

**Comment by jbartece** [comment]
@janinko Why do you think we should return JSON here? E.g, for build and alignment logs we return StreamingOutput.
If you all think it is better, then OK. I'm just curious why do you think it is better.

**Comment by alexcreasy** [comment]
@jbartece I admit it looks a little weird, but I agree with Honza, IMO the primary wire format of our API is JSON so we should return responses in JSON unless there is a good reason not to (e.g. a whole log file). 

The problem is, if you switch the method return type to a String, the content-type has to change from application/json to text/plain, I assume RESTEasy will automagically fix the headers for you.

In general it just places a burden on the client to remember there are different content types for different endpoints, the UI has quite a lot of interceptors and automatic code that assume JSON so I have to put some manual exceptions in for text/plain.

**Comment by alexcreasy** [comment]
Just to add that for a log file, it's a large stream of plain text, it would be highly expensive for the client to parse that as JSON just to reconvert it back to plain/text... The client really wants to put that in a buffer and handle it as it comes. 

For structured data JSON is better since it just gets parsed, I can grab the String from the message easily as a string and work with it like any other variable.

**Comment by jbartece** [comment]
OK, let's make it application/json then.

**Comment by michalovjan** [comment]
> @michalovjan I don't think it makes nicer code - it just complicates it. Another thing is that more important is, if it makes sense from the API perspective. I don't think so.
> I propose to close this PR and just remove the CUSTOM_PROJECT_MANIPULATOR_PARAMETERS property.

@jbartece I was referencing to the changes in my comment that just make the code nicer, sorry for the confusion.

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildConfigurationEndpointTest.java (hunk 3 above)
Comment (dominikbrazdil) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildConfigurationEndpointTest.java]: I can't find anything similar to String.isNotEmpty(), hamcrest supports something like this assertThat(string, is(not(emptyString()))) but that does not seem any better to me

Comment (jbartece) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildConfigurationEndpointTest.java]: You must call it on the object returned by assertThat not String.
E.g.: assertThat("test").isNotEmpty();

File: dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java (hunk 2 above)
Comment (dominikbrazdil) [reply dto/src/main/java/org/jboss/pnc/dto/response/AlignmentParameters.java]: no idea, when does intellij add these annotations?

**Pull request merged**